### PR TITLE
Migrate Employee list 

### PR DESCRIPTION
**Pull request by sarjon**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Migrates Employees list part of page.
| Type?         | new feature
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Partially migrates #10502
| How to test?  | Access new list directly via url `/admin-dev/index.php/configure/advanced/employees`. This PR **migrates only list**, actions will be implemented in separate PR.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/10797)

### CODE REVIEW

**Comment by sarjon** [comment]
@matks list is ready for review. dont forget that actions (bulk and row) will be implemented in separate PR to keep current pr scope manageable. :)

File: src/Core/Grid/Query/Builder/EmployeeQueryBuilder.php [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+                continue;
+            }
+
+            if ('active' === $filterName) {
+                $queryBuilder->andWhere('e.active = :active');
+                $queryBuilder->setParameter('active', $filterValue);
+                continue;
+            }
+
+            $queryBuilder->andWhere("$filterName LIKE :$filterName");
```
Comment (matks) [inline src/Core/Grid/Query/Builder/EmployeeQueryBuilder.php]: I am worried for possible SQL injections by using `"$filterName"` statement
Can we protect this field ?

File: src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/EmployeeController.php [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+     * @AdminSecurity("is_granted(['read', 'update', 'create', 'delete'], request.get('_legacy_controller'))")
+     *
+     * @param Request $request
+     *
+     * @return RedirectResponse
+     */
+    public function searchAction(Request $request)
+    {
+        $definitionFactory = $this->get('prestashop.core.grid.definition.factory.employee');
+        $emailLogsDefinition = $definitionFactory->getDefinition();
```
Comment (matks) [inline src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/EmployeeController.php]: Weird variable name `$emailLogsDefinition` :D
Copy-paste 😉 ?

**Review by matks — CHANGES_REQUESTED** [review]

File: src/Core/Grid/Query/Builder/EmployeeQueryBuilder.php (hunk 1 above)
Comment (PierreRambaud) [reply src/Core/Grid/Query/Builder/EmployeeQueryBuilder.php]: You're right, the best way is to defined which fields we want to use here

**Review by matks — DISMISSED** [review]

File: src/Core/Grid/Query/Builder/EmployeeQueryBuilder.php [hunk 3]
```diff
@@ -138,7 +138,20 @@ private function getEmployeeQueryBuilder(SearchCriteriaInterface $searchCriteria
      */
     private function applyFilters(QueryBuilder $queryBuilder, array $filters)
     {
+        $allowedFilters = [
```
Comment (PierreRambaud) [inline src/Core/Grid/Query/Builder/EmployeeQueryBuilder.php:141]: :+1:

**Comment by marionf** [comment]
@sarjon 

The delete option & the checkbox shouldn't appear for the superadmin employee created during install

capture d ecran_379

**Comment by sarjon** [comment]
@marionf you are right, but this PR does not migrate these actions, the correct display of actions will be migrated in separate PR. :) this one is about list and sorting/filtering.

**Comment by sarjon** [comment]
@matks rebased. :+1:

**Review by matks — APPROVED** [review]

**Pull request merged**